api/user/service.go

```diff
@@ -17,25 +18,42 @@ type service struct {
 	store store.Store
 }
 
-func NewService(store store.Store) Service {
-	return &service{store}
+type Invalid struct {
```

Comment:
My biggest concern is about the naming. This seems to be not clear about what is indeed "invalid". I think if it was named 'InvalidField' it'd be more meaningful.